src/app/courses-page/courses-page.component.ts

```diff
@@ -1,19 +1,23 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
 
 import { Course } from '@app/models';
 import { FilterPipe } from './pipes';
 import { CoursesService } from './services';
 
+import { Subscription } from 'rxjs';
+
 @Component({
   selector: 'app-courses-page',
   templateUrl: './courses-page.component.html',
   styleUrls: ['./courses-page.component.scss']
 })
-export class CoursesPageComponent {
+export class CoursesPageComponent implements OnDestroy {
   searchedCourses: Course[];
 
   get courses() {
-    return this.coursesService.getCourses();
+    let coursesList = [];
+    this.coursesSub = this.coursesService.getCourses().subscribe(courses => coursesList = courses);
```

Comment:
this will create new subscription each time which will result into memory leaks

Reply:
removed Observables